install.ps1

```diff
@@ -13,81 +14,291 @@ param (
 
-$name="dagger"
-$base="https://dl.dagger.io"
+function Get-PathMessage {
+    return $AddToPath ? "The dagger.exe has been added to your PATH" : "Please add dagger.exe to your PATH in order to use it"
+}
 
-function execute {
-    $url = base_url
-    $filename = tarball
-    $url = $url + "/" + $filename
-    write-host $url
-    if ($InteractiveInstall) {
-        Pause
+# This function prompts the user for a download location and validates it.
+function Get-DownloadPath {
+    $defaultPath = $DownloadPath
+    while ($true) {
+        $inputPath = Read-Host -Prompt "Enter the download location or leave empty and hit Enter for default ($defaultPath)"
+        if ([string]::IsNullOrWhiteSpace($inputPath)) {
+            return $defaultPath
+        } elseif (Test-Path $inputPath -IsValid) {
+            return $inputPath
+        } else {
+            Write-Host "Invalid path: $inputPath. Please enter a valid path."
+        }
     }
+}
 
-    Invoke-WebRequest -Uri $url -OutFile $env:temp/$filename -ErrorAction Stop
-    Expand-Archive -Path $env:temp/$filename -DestinationPath $InstallPath -Force -ErrorVariable ProcessError;
-    If ($ProcessError) {
-@"
----------------------------------------------------------------------------
-Whoops apparently we had an issue in unzipping the file, please check
-you have the right permission to do so and try to unzip manually the file.
-Currently we saved Dagger at your temp folder.
----------------------------------------------------------------------------
-"@
-        exit
-    } else {
-@"
+function Get-ProcessorArchitecture {
 
-Thank You for downloading Dagger!
+    $arch = $env:PROCESSOR_ARCHITECTURE
 
------------------------------------------------------
-Dagger has been saved at <YOUR HOME FOLDER>/dagger/
-Please add dagger.exe to your PATH in order to use it
-----------------------------------------------------
+    switch ($arch) {
+        "AMD64" {
+            return "amd64"
+        }
+        "ARM64" {
+            return "arm64"
+        }
+        "ARM" {
+            return "armv7"
+        }
+        default {
+            throw "Unsupported architecture: $arch"
+        }
+    }
+}
 
-"@
+function Get-ValidPath {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string]$Message,
+        [Parameter(Mandatory=$true)]
+        [string]$Default
+    )
+
+    $path = $null
+    while ($null -eq $path) {
+        $inputPath = Read-Host -Prompt $Message
+
+        if ([string]::IsNullOrWhiteSpace($inputPath)) {
+            return $Default
+        } else {
+            if (Test-Path $inputPath -IsValid) {
+                return $inputPath
+            } else {
+                Write-Host "Invalid path: $inputPath. Please enter a valid path."
+            }
+        }
     }
+}
 
+function Get-DownloadUrl {
+    $fileName = Get-FileName
+    return "https://github.com/dagger/dagger/releases/download/v${DaggerVersion}/${fileName}"
```

Comment:
This is a blocker - we should keep using `dl.dagger.io`.

Not quite sure what access issues you were seeing, do you have an example? Generally, this happens when the file doesn't exist on S3.

Reply:
I'll have another check on this in the evening.

Reply:
Fixed